### PR TITLE
fix: Update attributes `copy_protection_enabled`, `pit_enabled` and `encryption_at_rest_enabled` in the  resource `backup_compliance_policy` to be Optional

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description
Ticket: https://jira.mongodb.org/browse/CLOUDP-220487

`mongodbatlas_backup_compliance_policy` defines the `copy_protection_enabled`, `pit_enabled` and `encryption_at_rest_enabled` as required attributes when both the [API documentation](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Cloud-Backups/operation/updateDataProtectionSettings) and the [resource documentation](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/backup_compliance_policy) say the opposite.

Link to any related issue(s): #1800 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### I defined a new Acceptance tests
```hcl
Running tool: /Users/andrea.angiolillo/.asdf/shims/go test -timeout 300000s -run ^TestAccGenericBackupRSBackupCompliancePolicy_withoutOptionals$ github.com/mongodb/terraform-provider-mongodbatlas/internal/service/backupcompliancepolicy

ok  	github.com/mongodb/terraform-provider-mongodbatlas/internal/service/backupcompliancepolicy	15.761s
```

### Local test

- I installed the provider version 1.14.0
- I used the following TF conf
```hcl
resource "mongodbatlas_project" "atlas-project" {
  name   = "test-andrea"
  org_id = "XXXXXXXXXXX"
}

resource "mongodbatlas_backup_compliance_policy" "backup_policy" {
  project_id                 = mongodbatlas_project.atlas-project.id
  authorized_email           = "user@email.com"
  authorized_user_first_name = "ssss"
  authorized_user_last_name  = "sss"
  copy_protection_enabled    = false
  pit_enabled                = false
  encryption_at_rest_enabled = false

  restore_window_days = 7

  on_demand_policy_item {
    frequency_interval = 1
    retention_unit     = "days"
    retention_value    = 3
  }

  policy_item_hourly {
    frequency_interval = 1
    retention_unit     = "days"
    retention_value    = 7
  }

  policy_item_daily {
    frequency_interval = 1
    retention_unit     = "days"
    retention_value    = 7
  }

  policy_item_weekly {
    frequency_interval = 1
    retention_unit     = "weeks"
    retention_value    = 4
  }

  policy_item_monthly {
    frequency_interval = 1
    retention_unit     = "months"
    retention_value    = 12
  }
}
```

- Run `terraform apply`
```hcl
Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
```
- I upgraded the provider to the local changes
- I removed the optional fields from the tf conf
- I run `terraform apply`
```hcl
No changes. Your infrastructure matches the configuration.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

### CODE REVIEW

File: internal/service/backupcompliancepolicy/resource_backup_compliance_policy.go [hunk 1]
```diff
@@ -414,7 +417,7 @@ func resourceMongoDBAtlasBackupCompliancePolicyUpdate(ctx context.Context, d *sc
 
 	backupCompliancePolicyUpdate.AuthorizedEmail = d.Get("authorized_email").(string)
 
-	backupCompliancePolicyUpdate.AuthorizedUserFirstName = d.Get("authorized_user_fist_name").(string)
+	backupCompliancePolicyUpdate.AuthorizedUserFirstName = d.Get("authorized_user_first_name").(string)
```
Comment (andreaangiolillo) [inline internal/service/backupcompliancepolicy/resource_backup_compliance_policy.go:420]: [drive-by] This misspell was causing a crash during the update

```hcl
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-mongodbatlas plugin:

panic: interface conversion: interface {} is nil, not string
```

**Review by oarbusi — APPROVED** [review]
LGTM

**Review by AgustinBettati — APPROVED** [review]
LGTM, do you think we can adjust the PR title to include more detail of the adjusted attributes?
